### ISSUE TITLE
tour: sync.Mutex example works the same without mutex

### ISSUE THREAD

**Issue by pdobacz** [golang/tour#129]
Context: https://tour.golang.org/concurrency/9

The example behaves perfectly fine, even if I comment out the `Lock`/`Unlock`. To make the mutex required I needed to modify the `Inc` function:
```go
	c.mux.Lock()
	// Lock so only one goroutine at a time can access the map c.v.
	value := c.v[key]
	value += 1
	time.Sleep(time.Nanosecond)
	c.v[key] = value
	c.mux.Unlock()
```

(using the web hosted go-tour)

**Comment by ladlestein** [comment]
Concur

**Comment by bgadrian** [comment]
In a real env, with multi core, probably the program will crash if you comment the mutex. In the playground there are some CPU limitations.

**Comment by pdobacz** [comment]
> In a real env, with multi core, probably the program will crash if you comment the mutex. In the playground there are some CPU limitations.

True that, probably.

Still this makes this particular exercise quite insipid and might lead curious newcomers to false conclusions.

**Comment by pdobacz** [comment]
> Which lock/unlock pair did you commented?

definitely the one in `Inc`. I don't get a crash but the exact same result as with unmodified code (in the playground, of course). Funny.

**Comment by ALTree** [comment]
You're right, the concurrent map access check does not work on the playground. It's a best-effort thing.

I'd leave the code as it is. It doesn't seem worth it to change it just to tentatively trigger some data-races in the (very unusual) playground environment.

OTHO we could improve the accompanying text to explain better what a data-race is, and to stress that a program with a data-race in it is 1) not valid 2) not expected to work correctly.

**Comment by ALTree** [comment]
Just notice we re-enacted the same exact conversation they had at #35 

I guess I'll take this as a confirmation that a little more text would help.

**Comment by matthedude** [comment]
I too spent a few minutes wondering why there was no change when commenting out the mutexes - the playground environment limitations makes perfect sense but +1 for an explanation on the page.

**Comment by crisman** [comment]
Works now in playground, Mutex is needed.

In the current (early 2023 & go 1.20) playground commenting out `Lock()` and `Unlock()` in `Inc()` frequently generates a `fatal error: concurrent map writes` and rarely a number less than 1000.

**Comment by ALTree** [comment]
Thanks, closing here then.
